kpi/deployment_backends/kobocat_backend.py

```diff
@@ -505,6 +563,65 @@ def __get_submissions_in_xml(self, instances_ids=[]):
         if len(instances_ids) > 0:
             queryset = queryset.filter(id__in=instances_ids)
 
-        queryset = queryset.order_by("id")
+        # Sort
+        sort = params.get("sort") or sort
+        sort_key = sort.keys()[0]
+        sort_dir = int(sort[sort_key])  # -1 for desc, 1 for asc
+        queryset = queryset.order_by("{direction}{field}".format(
+            direction="-" if sort_dir < 0 else "",
+            field=sort_key
+        ))
+
+        # When using Mongo, data is already paginated, no need to do it with PostgreSQL too.
+        if not use_mongo:
+            offset = params.get("start")
+            limit = offset + params.get("limit")
+            queryset = queryset[offset:limit]
 
         return (lazy_instance.xml for lazy_instance in queryset)
+
+    @staticmethod
+    def __kobocat_proxy_request(kc_request, user=None):
+        """
+        Send `kc_request`, which must specify `method` and `url` at a minimum.
+        If the incoming request to be proxied is authenticated,
+        logged-in user's API token will be added to `kc_request.headers`
+
+        :param kc_request: requests.models.Request
+        :param user: User
+        :return: requests.models.Response
+        """
+        if not user.is_anonymous() and user.pk != settings.ANONYMOUS_USER_ID:
+            token, created = Token.objects.get_or_create(user=user)
+            kc_request.headers['Authorization'] = 'Token %s' % token.key
+        session = requests.Session()
+        return session.send(kc_request.prepare())
+
+    @staticmethod
+    def __prepare_as_drf_response_signature(requests_response):
+        """
+        Prepares a dict from `Requests` response.
+        Useful to get response from `kc` and use it as a dict or pass it to
+        DRF Response
+        """
+
+        prepared_drf_response = {}
+
+        try:
+            prepared_drf_response["content_type"] = requests_response.headers.get("Content-Type")
+            prepared_drf_response["headers"] = {
+                "Content-Language": requests_response.headers.get("Content-Language")
+            }
+        except (KeyError, AttributeError):
+            pass
+
+        prepared_drf_response["status"] = requests_response.status_code
+
+        try:
+            prepared_drf_response["data"] = json.loads(requests_response.content)
+        except ValueError as e:
+            if not requests_response.status_code == status.HTTP_204_NO_CONTENT:
+                raise Exception(str(e))
```

Comment:
I have changed the code a little bit. 
The error is returned in the response. `FE` can expose the message to users. 
For example, if submission deletion fails, nothing happens. It would be better to display a message to users. (cc @magicznyleszek)